NetworkingTests/Tests/SessionManagerSpec.swift

```diff
@@ -0,0 +1,473 @@
+                }}
+                
+            }
+            
+            context("when bootstraps and there is no session") {
+                
+                beforeEach {
+                    keychainService = KeychainServiceMock()
+                    sessionManager = SessionManager(keychainService: keychainService)
+                }
+                
+                it("sends none in user signal") { waitUntil { done in
+                    sessionManager.userSignal.observeValues {
+                        switch $0 {
+                        case .none: done()
+                        case .some: fail()
+                        }
+                    }
+                    
+                    sessionManager.bootstrap()
+                }}
+                
+            }
+            
+        }
+        
+        describe("#login") {
+            
+            describe("when there is a current session") {
+                
+                beforeEach {
+                    keychainService = KeychainServiceMock()
+                    keychainService.set(value: UserMock().sessionToken!,
+                                        forKey: SessionManagerSpec.CurrentSessionTokenPersistanceKey)
+                    sessionManager = SessionManager(keychainService: keychainService)
+                    sessionManager.bootstrap()
+                }
+                
+                it("throws an assertion") {
+                    expect(sessionManager.login(user: UserMock())).to(throwAssertion())
+                }
+                
+            }
+            
+            describe("when there is no current session") {
+                
+                beforeEach {
+                    keychainService = KeychainServiceMock()
+                    sessionManager = SessionManager(keychainService: keychainService)
+                    sessionManager.bootstrap()
+                }
+                
+                it("sends true in session signal") { waitUntil { done in
+                    sessionManager.sessionSignal.observeValues {
+                        switch $0 {
+                        case true: done()
+                        case false: fail()
+                        }
+                    }
+                    
+                    sessionManager.login(user: UserMock())
+                }}
+                
+                it("sends the authenticated user in user signal") { waitUntil { done in
+                    sessionManager.userSignal.observeValues {
+                        switch $0 {
+                        case .none: fail()
+                        case .some(let user):
+                            expect(user.sessionToken).to(equal(UserMock().sessionToken))
+                            done()
+                        }
+                    }
+                    
+                    sessionManager.login(user: UserMock())
+                }}
+                
+                it("returns the session token") {
```

Comment:
the login doesn't return anything. I would move this to when you describe the `#currentUser` property.